static/js/hackathons.js

```diff
@@ -1,9 +1,100 @@
+
+function flipCard(hackathonId) {
+    const card1 = document.getElementById(`${hackathonId}-card-front`);
+    const card2 = document.getElementById(`${hackathonId}-card-back`);
+    card1.classList.toggle("flipped")
+    card2.classList.toggle("flipped")
+}
+
 function saveHackathon(event, hackathonId) {
     event.preventDefault();
     var xhr = new XMLHttpRequest();
-    xhr.open('POST', `${hackathonId}/save`, true);
+    const saveButton = document.getElementById(`${hackathonId}-save`);
+    if (saveButton.classList.contains('bg-[#e5462e]')) {
+        saveButton.classList.remove('bg-[#e5462e]');
+        saveButton.classList.remove('text-white');
+        saveButton.classList.add('bg-white');
+        saveButton.textContent = "Save";
+    }
+    else {
+        saveButton.classList.add('bg-[#e5462e]');
+        saveButton.classList.add('text-white');
+        saveButton.classList.remove('bg-white');
+        saveButton.textContent = "Saved";
+    }
+    xhr.open('POST', `/hackathons/${hackathonId}/save`, true);
     xhr.setRequestHeader('Content-Type', 'application/json');
     xhr.setRequestHeader('X-CSRFToken', getCookie('csrftoken'));
+    xhr.send(JSON.stringify({}));
+}
+
+function cancelLoginPop(){
+    const popup = document.getElementById('login-popup');
+    popup.classList.add('hidden');
+    popup.classList.remove('flex');
+}
+
+function addLoginPop(){
+    const popup = document.getElementById('login-popup');
+    popup.classList.remove('hidden');
+    popup.classList.add('flex');
+}
+
+
+function voteHackathon(event, hackathonId, vote_state) {
+    event.preventDefault();
+    var xhr = new XMLHttpRequest();
+
```

Comment:
do you need a CSRF token here?

Reply:
yea, I have it at the bottem with xhr.setRequestHeader('X-CSRFToken', getCookie('csrftoken'));

Reply:
Missed that